pilot/cmd/pilot-agent/status/ready/probe.go

```diff
@@ -108,3 +119,21 @@ func (p *Probe) checkServerInfo() error {
 
 	return nil
 }
+
+// pingListeners checks to ensure that Envoy is actually listenening on the port.
+func (p *Probe) pingVirtualListeners() error {
+
+	// Check if traffic capture ports are actually listening.
+	vports, err := util.GetVirtualListenerPorts(p.LocalHostAddr, p.AdminPort)
+	if err != nil {
+		return err
+	}
+	for _, vport := range vports {
+		_, err := net.Dial("tcp", fmt.Sprintf("%s:%d", p.LocalHostAddr, vport))
```

Comment:
Need to close the connection explicitly.

Reply:
Oh. Yeah. thanks. Fixed now.